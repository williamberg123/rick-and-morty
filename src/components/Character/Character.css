.Character {
    position: relative;
    display: flex;
    flex-direction: column;
    padding: 5px;
    border-radius: 10px;
    box-shadow: 3px 3px 10px rgb(138, 138, 138);
    transition: 0.3s;
    font-weight: bold;
    aspect-ratio: 1/1;
}

.Character:hover {
    transform: scale(0.95);
}

.Character img {
    width: 100%;
    z-index: 2;
    border-radius: 10px;
}

.shadow {
    background-color: rgba(0, 0, 0, 0.4);
    position: absolute;
    width: calc(100% - 10px);
    height: calc(100% - 10px);
    z-index: 2;
    border-radius: 10px;
}

.Character-info {
    width: calc(100% - 15px);
    position: absolute;
    bottom: 5px;
    left: 15px;
    z-index: 3;
}

.Character-info h3 {
    color: red;
    font-size: 1rem;
}

.Character p {
    color: rgb(233, 233, 233);
}

@media (max-width: 700px) {
    .Character h3 {
        font-size: 0.7rem;
    }

    .Character p {
        font-size: 0.7rem;
    }
}

@media (max-width: 450px) {
    .Character {
        padding: 3px;
    }

    .shadow {
        width: calc(100% - 6px);
        height: calc(100% - 6px);
    }

    .Character-info {
        left: 10px;
        bottom: 0;
    }
}