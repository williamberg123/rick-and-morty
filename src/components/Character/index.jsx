import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import Image from '../Image';

import './style.css';

function Character({ id, name, species, imageUrl }){
    return(
        <Link to={`/rick-and-morty/personagem?id=${id}`}>
            <div className="Character" data-testid="CharacterDiv">
                <Image
                    imageUrl={imageUrl}
                    altName={name}
                />
                <div className="shadow"></div>
                <div className="Character-info">
                    <h3>{name.toUpperCase()}</h3>
                    <p>{species}</p>
                </div>
            </div>
        </Link>
    );
}

Character.propTypes = {
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    imageUrl: PropTypes.string.isRequired
};

export default memo(Character);