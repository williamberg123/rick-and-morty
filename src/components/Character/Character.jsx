import React from 'react';
import PropTypes from 'prop-types';

import './Character.css';

export default function Character({ name, species, imageUrl }){
    return(
        <div className="Character">
            <img src={imageUrl} alt={name} />
            <div className="shadow"></div>
            <div className="Character-info">
                <h3>{name.toUpperCase()}</h3>
                <p>{species}</p>
            </div>
        </div>
    );
}

Character.propTypes = {
    name: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    imageUrl: PropTypes.string.isRequired
};