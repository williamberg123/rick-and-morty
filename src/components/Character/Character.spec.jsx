import React from 'react';
import { render, screen } from '@testing-library/react';
import Character from './Character';

describe('<Character />', () => {
    it('should render him elements correctly', () => {
        expect.assertions(3);

        render(<Character name="Rick Sanchez" species="Human" imageUrl="img/img.jpg" />);

        expect(screen.getByRole('img', { name: /rick sanchez/i })).toBeInTheDocument();
        expect(screen.getByRole('heading', { name: /rick sanchez/i })).toBeInTheDocument();
        expect(screen.getByText('Human')).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        render(<Character name="Rick Sanchez" species="Human" imageUrl="img/img.jpg" />);
        expect(screen.getByTestId('CharacterDiv')).toMatchSnapshot();
    });
});

// Testado