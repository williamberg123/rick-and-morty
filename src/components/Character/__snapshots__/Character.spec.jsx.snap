// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Character /> should match snapshot 1`] = `
<div
  class="Character"
  data-testid="CharacterDiv"
>
  <img
    alt="Rick Sanchez"
    src="img/img.jpg"
  />
  <div
    class="shadow"
  />
  <div
    class="Character-info"
  >
    <h3>
      RICK SANCHEZ
    </h3>
    <p>
      Human
    </p>
  </div>
</div>
`;
